server/network/aoprotocol.py

```diff
@@ -38,6 +39,7 @@ class ProtocolError(Exception):
 
 class AOProtocol(asyncio.Protocol):
     """The main class that deals with the AO protocol."""
+    last_message_char_id: int = -1
```

Comment:
This should go in the area class

Reply:
Gotcha. Thanks!

Reply:
I've added them to the area and the protocol checks to see if it's allowed @oldmud0